### PR TITLE
Do not raise up a dead body!

### PR DESCRIPTION
**Pull request by JesterX666**
Fixes Dead/crit players get back up when stun is over #2407

### CODE REVIEW

**Review by daalta — COMMENTED** [review]
Looks good, trying it in the editor now.

**Comment by fomalsd** [comment]
Scr 2020-02-01 в 1 19 33
does the job.
one thing I've noticed though is that player also gets up in crit, while not supposed to.
so maybe extend that fix to crit players, too?

**Comment by daalta** [comment]
@fomalsd Hmmm... Maybe? Instead of checking if a player is dead we could check if they're not ConsciousState.CONSCIOUS. Though I feel like that might introduce problems with players not being able to stand up once the stun wears off.

We could make a separate issue for that. @JesterX666 's fix is probably good enough to fix dead players standing up..?

**Comment by fomalsd** [comment]
#2407 clearly reads "Dead/Crit..." while this fix only addresses dead ones.
I think there's an event being invoked when conscious state is changed (that eventually updates upright state, too), so checking for ConsciousState.CONSCIOUS makes sense to me

File: UnityProject/Assets/Scripts/Tilemaps/Behaviours/Objects/RegisterPlayer.cs [hunk 1]
```diff
@@ -195,7 +195,13 @@ private void ServerRemoveStun()
 	{
 		var oldVal = IsSlippingServer;
 		IsSlippingServer = false;
-		SyncIsLayingDown(false);
+
+		// Do not raise up a dead body
+		if (playerScript.playerHealth.ConsciousState != ConsciousState.DEAD)
```
Comment (fomalsd) [inline UnityProject/Assets/Scripts/Tilemaps/Behaviours/Objects/RegisterPlayer.cs]:
```suggestion
		if (playerScript.playerHealth.ConsciousState == ConsciousState.CONSCIOUS)
```

**Comment by JesterX666** [comment]
Oh!  I forgot about "crit".  I will do it as soon this evening.

**Comment by JesterX666** [comment]
Yup, the == Concsious does take care of not getting up while "crit"
Thanks @fomalsd

**Review by fomalsd — APPROVED** [review]

**Pull request merged**